### PR TITLE
sort commits before finding commit by ref

### PR DESCRIPTION
**Pull request by johnnyaug**

### CODE REVIEW

File: graveler/ref/manager.go [hunk 1]
```diff
@@ -214,6 +214,7 @@ func (m *Manager) GetCommitByPrefix(ctx context.Context, repositoryID graveler.R
 					SELECT id, committer, message, creation_date, parents, meta_range_id, metadata
 					FROM graveler_commits
 					WHERE repository_id = $1 AND id >= $2
+					ORDER BY id
```
Comment (arielshaqed) [inline graveler/ref/manager.go:217]: I don't understand this change (but, to the PR's defence, I had several rounds with @ozkatz around this very SQL query).  This query asks for at most 2 rows; if there is more than one row then l.235 returns `ErrRefAmbiguous`.  So... AFAIU either `GetCommitByPrefix` fetches and returns a single row or it returns an error.  As such order *cannot* matter.

**Review by arielshaqed — CHANGES_REQUESTED** [review]
Not sure this PR has any effect, but it might make things slower.  @ozkatz perhaps the comment would be more noticeable if be placed *after* the query?

**Comment by johnnyaug** [comment]
The ordering is performed before the limit. Without the order by clause,
the query may return any 2 rows with keys greater than the prefix

File: graveler/ref/manager.go (hunk 1 above)
Comment (johnnyaug) [reply graveler/ref/manager.go:217]: The ordering is performed before the limit. Without the order by clause,
the query may return any 2 rows with keys greater than the prefix

File: graveler/ref/manager.go [hunk 2]
```diff
@@ -213,7 +213,7 @@ func (m *Manager) GetCommitByPrefix(ctx context.Context, repositoryID graveler.R
 		err := tx.Select(&records, `
 					SELECT id, committer, message, creation_date, parents, meta_range_id, metadata
 					FROM graveler_commits
-					WHERE repository_id = $1 AND id >= $2
+					WHERE repository_id = $1 AND id LIKE $2 || '%'
```
Comment (arielshaqed) [inline graveler/ref/manager.go:216]: This is the important part of the fix, right?  LGTM!

**Review by arielshaqed — APPROVED** [review]
Nice!  As discussed f2f, the worry is that a database (let's call it PostgreSQeviL) might always return the last 2 commit digests, neither of which has the prefix, when there is precisely one prefix.
Maybe add a test?

Also, I think maybe "addressProvider" is from another change?  So I did *not* review it, I'm assuming you will undo before pushing.

THANKS!

**Comment by johnnyaug** [comment]
@arielshaqed actually the addressProvider was added in order to be able to test this. I'm actually not sure it's worth it, but I wanted to add a test. WDYT?

**Review by arielshaqed — APPROVED** [review]
OK, now I see it!  That's actually really neat, and it allows us (you) to test that things work.  THANKS!

**Pull request merged**